pkg/oci/oci.go

```diff
@@ -0,0 +1,316 @@
+	if err != nil {
+		return ocispec.Manifest{}, fmt.Errorf("fetch manifest: %w", err)
+	}
+	defer reader.Close()
+	bytes, err := io.ReadAll(reader)
+	if err != nil {
+		return ocispec.Manifest{}, fmt.Errorf("read manifest: %w", err)
+	}
+	var manifest ocispec.Manifest
+	err = json.Unmarshal(bytes, &manifest)
+	if err != nil {
+		return ocispec.Manifest{}, fmt.Errorf("unmarshal manifest: %w", err)
+	}
+	return manifest, nil
+}
+
+func GetContentFromDescriptor(imageStore oras.ReadOnlyTarget, desc ocispec.Descriptor) ([]byte, error) {
+	reader, err := imageStore.Fetch(context.Background(), desc)
+	if err != nil {
+		return nil, fmt.Errorf("fetch descriptor: %w", err)
+	}
+	defer reader.Close()
+	bytes, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, fmt.Errorf("read descriptor: %w", err)
+	}
+	return bytes, nil
+}
+
+func GetDefinition(target oras.Target, authOpts *AuthOptions, image string) ([]byte, error) {
+	err := PullIfNotExist(target, authOpts, image)
+	if err != nil {
+		return nil, fmt.Errorf("pull image: %w", err)
+	}
+	index, err := GetImageListDescriptor(target, image)
+	if err != nil {
+		return nil, fmt.Errorf("get image list descriptor: %w", err)
+	}
+	manifest, err := GetHostArchManifest(target, index)
+	if err != nil {
+		return nil, fmt.Errorf("get arch manifest: %w", err)
+	}
+	definition, err := GetContentFromDescriptor(target, manifest.Config)
+	if err != nil {
+		return nil, fmt.Errorf("get definition from descriptor: %w", err)
+	}
+
+	return definition, nil
+}
+
+func GetEbpfProgram(target oras.Target, authOpts *AuthOptions, image string) ([]byte, error) {
+	err := PullIfNotExist(target, authOpts, image)
+	if err != nil {
+		return nil, fmt.Errorf("pull image: %w", err)
+	}
+	index, err := GetImageListDescriptor(target, image)
+	if err != nil {
+		return nil, fmt.Errorf("get image list descriptor: %w", err)
+	}
+	manifest, err := GetHostArchManifest(target, index)
+	if err != nil {
+		return nil, fmt.Errorf("get arch manifest: %w", err)
+	}
+	if len(manifest.Layers) != 1 {
+		return nil, fmt.Errorf("expected exactly one layer, got %d", len(manifest.Layers))
+	}
+	definition, err := GetContentFromDescriptor(target, manifest.Layers[0])
+	if err != nil {
+		return nil, fmt.Errorf("get ebpf program from descriptor: %w", err)
+	}
+
+	return definition, nil
+}
+
+func GetTagFromImage(image string) (string, error) {
+	repo, err := reference.Parse(image)
+	if err != nil {
+		return "", fmt.Errorf("parse image %q: %w", image, err)
+	}
+	tagged, ok := repo.(reference.Tagged)
+	if !ok {
+		return "latest", nil
+	}
+	return tagged.Tag(), nil
+}
+
+func GetRepositoryFromImage(image string) (string, error) {
+	repo, err := reference.Parse(image)
+	if err != nil {
+		return "", fmt.Errorf("parse image %q: %w", image, err)
+	}
+	if named, ok := repo.(reference.Named); ok {
+		return named.Name(), nil
+	}
+	return "", fmt.Errorf("image has to be a named reference")
+}
+
+func NormalizeImage(image string) (string, error) {
```

Comment:
```suggestion
func NormalizeImageName(image string) (string, error) {
```

I think this just normalizes the name, right?